sympy/matrices/expressions/matadd.py

```diff
@@ -52,11 +54,11 @@ def __new__(cls, *args, evaluate=False, check=None, _sympify=True):
                 "Passing check to MatAdd is deprecated and the check argument will be removed in a future version.",
                 deprecated_since_version="1.11",
                 active_deprecations_target='remove-check-argument-from-matrix-operations')
-        if check in (True, None):
+        if check:
             if not any(isinstance(i, MatrixExpr) for i in args):
                 return Add.fromiter(args)
             validate(*args)
```

Comment:
The type check logic (`MatrixExpr` vs `Expr`)is inside this, so I'd give a reminder that I may need to distinguish and pull out the type checks in the future